File: kotlin/programmers/lv2/멀리뛰기.kt
class Solution {
  fun solution(n: Int): Long {
      val dp = LongArray(2001)
      
      dp.forEachIndexed { index, _ ->  
          if (index <= 2) {
              dp[index] = index.toLong()
              return@forEachIndexed
          }
          
          dp[index] = (dp[index - 1] + dp[index - 2]) % 1234567
      }
      
      return dp[n]
  }
}

File: javascript/problems/programmers/lv.2/멀리뛰기.js
function solution(n) {
  const dp = new Array(2001).fill(0);

  dp.forEach((v, index) => {
    if ([0, 1, 2].some((v) => v === index)) {
      dp[index] = BigInt(index);

      return;
    }

    dp[index] = BigInt(dp[index - 1]) + BigInt(dp[index - 2]);
  });

  return Number(dp[n] % 1234567n);
}
